webclient/app/src/app/commons/appTemplateDialog/AppTemplateAuthDialog.jsx

```diff
@@ -0,0 +1,144 @@
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import { useTranslation } from 'react-i18next';
+import AppTemplateRest from "../../services/AppTemplateRest";
+import { CloudSync, Close } from "@mui/icons-material";
+import ErrorAlert from "../alert/ErrorAlert";
+import AppTemplateDialogStyles from "./AppTemplateDialogStyles";
+import NotificationDialog from "../alert/NotificationDialog";
+import SimpleValidatedTextField from "../validatedTextField/SimpleValidatedTextField";
+import RegexConfig from "../../../regexConfig";
+
+function AppTemplateAuthDialog(props) {
+    const { appTemplate, handleRefresh, setOpenSuccessDialog } = props;
+    const appTemplateDialogStyles = AppTemplateDialogStyles();
+    const [hasFormError, setHasFormError] = React.useState(false);    
+    const [downloadRequestData, setDownloadRequestData] = useState({ "appTemplateId": null, "username": "", "password": "" });
+    const [alert, setAlert] = useState({"open":false, "title": "ERROR", "message": ""});
+    const { t } = useTranslation();
+
+    const appTemplateRest = new AppTemplateRest();
+
+    const [openAuthDialog, setOpenAuthDialog] = useState(false);
+  
+
+    const handleChange = (event) => {
+        const { name, value } = event.target;
+        let downloadRequestDataNew = { ...downloadRequestData };
+        downloadRequestDataNew[name] = value;
+        setDownloadRequestData(downloadRequestDataNew);
+    }
+
+    const handleLogin = () => {
+        if (appTemplate.credentialsRequired) {
+            handleAlertClose();
+            setOpenAuthDialog(true);
+        } else {
+            handleAppTemplateReload();
+        }
+    }
+
+    const handleAppTemplateReload = () => {
+        if(hasFormError && appTemplate.credentialsRequired) {
+            return;
+        }
+        downloadRequestData.appTemplateId = appTemplate.id;
+        appTemplateRest.updateTemplates(downloadRequestData).then(() => {
+            handleRefresh();
+            setOpenAuthDialog(false);
+            setOpenSuccessDialog(true);
+        }).catch(err => {
+            setAlert({"open":true, "title": t("alert.error"), "message": t(err.response.data.messageKey)});
+            console.log(err.response.data);
+        });
+    }
+
+    const handleAlertClose = () => {
+        setAlert({"open":false, "title": "ERROR", "message": ""});
+    }
+
+    const onClose = () => {
+        setDownloadRequestData({ "appTemplateId": null, "username": "", "password": "" });
+        setOpenAuthDialog(false);
+    }
+
+    useEffect(() => {
+        if (!downloadRequestData) {
+            return;
+        }
+        let hasError = false;
+
+        if (!RegexConfig.appTemplateAuthUser.test(downloadRequestData.username)) {
+            hasError = true;
+        }
+        if (!RegexConfig.appTemplateAuthPassword.test(downloadRequestData.password)) {
+            hasError = true;
```

Comment:
Why does this have a password check?

Reply:
is a must because off command line call of the git command.